File: BaekJoon/BruteForce/1018.py
N, M = map(int, input().split())

temp = []
case = []

for _ in range(N):
    temp.append(input())

for i in range(0, N-7):
    for j in range(0, M-7):
        c1 = 0
        c2 = 0

        for a in range(8):
            for b in range(8):
                if ((a+b)%2 == 0 and temp[a+i][b+j] == "B") or ((a+b)%2 == 1 and temp[a+i][b+j] == "W"):
                    c1 += 1
                
        for a in range(8):
            for b in range(8):
                if ((a+b)%2 == 1 and temp[a+i][b+j] == "B") or ((a+b)%2 ==0 and temp[a+i][b+j] == "W"):
                    c2 += 1
        case.append(min(c1,c2))


print(min(case))

File: BaekJoonSwift/BaekJoonSwift/Bruth Force/1018.swift
import Foundation

let input = readLine()!.split(separator: " ").map { Int($0)! }
var chess = [[String]]()
var results = [Int]()

for _ in 0..<input[0] {
    let temp = Array(readLine()!).map { String($0) }
    chess.append(temp)
}


for i in 0...input[0]-8 {
    for j in 0...input[1]-8 {
        var a = 0 // start with B
        var b = 0 // start with W
        
        for r in 0..<8 {
            for c in 0..<8 {
                if (r+c) % 2 == 0 && chess[i+r][j+c] == "W" || (r+c) % 2 == 1 && chess[i+r][j+c] == "B" {
                    a += 1
                }
            }
        }
        
        for r in 0..<8 {
            for c in 0..<8 {
                if (r+c) % 2 == 0 && chess[i+r][j+c] == "B" || (r+c) % 2 == 1 && chess[i+r][j+c] == "W"{
                    b += 1
                }
            }
        }
        
        results.append(min(a,b))
    }
}

print(results.sorted().first!)
